examples/sqlite/test/tests.ts

```diff
@@ -2,28 +2,525 @@ import * as dns from 'node:dns';
 dns.setDefaultResultOrder('ipv4first');
 
 import { Test } from 'azle/test';
+import { execSync } from 'child_process';
 
 export function getTests(canisterId: string): Test[] {
     const origin = `http://${canisterId}.localhost:8000`;
 
     return [
         {
-            name: 'load db',
-            test: async () => {
-                try {
-                    const response = await fetch(origin, {
-                        method: 'POST'
-                    });
-                    const responseText = await response.text();
-
-                    return {
-                        Ok: responseText === 'Hello world'
-                    };
-                } catch (error: any) {
-                    return {
-                        Err: error
-                    };
-                }
+            name: '/init-called true',
+            test: async () => {
+                const response = await fetch(`${origin}/init-called`);
+                const responseJson = await response.json();
+
+                return {
+                    Ok: responseJson === true
+                };
+            }
+        },
+        {
+            name: '/post-upgrade-called false',
+            test: async () => {
+                const response = await fetch(`${origin}/post-upgrade-called`);
+                const responseJson = await response.json();
+
+                return {
+                    Ok: responseJson === false
+                };
+            }
+        },
+        ...usersTestsBeforeBatch(origin),
+        ...postsTestsBeforeBatch(origin),
+        {
+            name: '/users/batch/499',
+            test: async () => {
+                const response = await fetch(`${origin}/users/batch/499`, {
```

Comment:
Could we do a constant instead of a magic number, and could we do something like FINAL_TOTAL - 1 or something to better illustrate that we added one in the before batch and we are filling it all the way up to a set number? Alternatively have the final total be BATCH_ADD_USER_COUNT + 1. (thought with 500 being the final it does feel more like a FINAL_TOTAL - 1 situation, while 501 would feel more like a BATCH_ADD_USER_COUNT + 1 situation)

Reply:
Did something like this